eventing(samples): update Node.js CloudEvents ContainerSource example (#4039)
knative/docs: docs/developer/eventing/sources/creating-event-sources/writing-event-source-easy-way/README.md

Create the project and add the dependencies:

```bash
npm init -y
npm install cloudevents got --save
```

## Using ContainerSource

A ContainerSource creates a container for your event source image and manages this container.

> The sink URL where events are posted will be made available to the application through the `K_SINK` environment variable by the ContainerSource.

Reviewer: ```suggestion
The sink URL where events are posted is made available to the application through the `K_SINK` environment variable, provided by the ContainerSource.
```

  // Emits the 'cloudevent' Node.js event application-wide
  event.emit();
}, 1000);

// Create a function that can post an event
const emit = emitterFor(event => {
  got.post(K_SINK, event)
    .then(response => {
      console.log('Event posted successfully');
      console.log(response.data);
    })
    .catch(err => {
      console.log('Error during event post');
      console.error(err);
    });
  });

// Send the CloudEvent any time a Node.js 'cloudevent' event is emitted
Emitter.on('cloudevent', emit);

registerGracefulExit();

function registerGracefulExit() {
  process.on('exit', logExit);
  //catches ctrl+c event
  process.on('SIGINT', logExit);
  process.on('SIGTERM', logExit);
  // catches 'kill pid' (for example: nodemon restart)
  process.on('SIGUSR1', logExit);
  process.on('SIGUSR2', logExit);
}

function logExit(message = 'Exiting...') {
  // Handle graceful exit
  console.log(message);
  process.exit();
}
```

```dockerfile
# File - Dockerfile

FROM node:16
WORKDIR /usr/src/app
COPY package*.json ./
RUN npm install
COPY . .
EXPOSE 8080
CMD [ "node", "index.js" ]
```

### Procedure

> Before publishing the ContainerSource, we must build the application
> image, and push it to a container registry accessible by our cluster.

Reviewer 2: ```suggestion
Before publishing the ContainerSource, you must build the application
image, and push it to a container registry that your cluster can access.
```
